[CS2113T-W13-1] revisED

syncode98 opened the pull request:
revisED helps students to take notes for the subjects that they are studying. It is optimised for CLI users so that the students can quickly add their notes for the subjects that they are studying,

syncode98 committed:
Refactor the printUpcomingTasks method

The method has been refactored to ensure that
the list of tasks are printed out correctly.

syncode98 committed:
Change the default LocalDateTime set to Task

The default LocalDateTime of tasks has been
changed to LocalDateTime.max .This is to prevent
any NullPointerExceptions.

syncode98 committed:
Remove the RepeatedDateTimeException

The exception has been removed to
allow users to add tasks with same deadlines.

sugandha929 committed:
Resolve suggestions for UG

syncode98 committed:
Merge pull request #82 from sugandha929/master

Update UG and DG

syncode98 committed:
Merge pull request #83 from CookieHoodie/storage-imp

Improve on implementation of task methods

syncode98 committed:
Merge branch 'master' into Quiz_dateTime

syncode98 committed:
Edit method in Ui class

The printStartSubjectQuiz method
in Ui class has been changed to
print the correct output.

syncode98 committed:
Edit the UserGuide

The icons have been replaced with
section headers instead.

syncode98 committed:
Update Developer Guide

Included information regarding the quiz package.

Rectify checkstyle test errors.

sugandha929 committed:
Merge pull request #84 from syncode98/Quiz_dateTime

Quiz_dateTime

sugandha929 committed:
Merge pull request #5 from AY2021S1-CS2113T-W13-1/master

pull changes

sugandha929 committed:
Update DG

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+present in the `SubjectQuiz` class . 
+
+The application now returns to the `startQuiz` method and the current score of the quiz is set to zero. Subsequently, the
+application begins printing the questions from the flashcards and checks the answer that you provide. If you want to stop the 
+quiz, use the `stop` command. The application will then print out the score that you obtain. If you complete the quiz
+then the application not only prints the score, but it also prints the incorrect answers from the quiz.
+
+The following diagram shows how you can initiate the quiz for a subject.
```
Should this be a solid arrow line instead? 

![Capture1](https://user-images.githubusercontent.com/57391861/97386709-5d923980-190f-11eb-9733-bd9f6ba581de.PNG)

flsqqq replied:
Should the person using the app be referred to as "the user" instead of "you"?

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+This class also stores the data in a cascading manner, meaning that when a call to save topics is made, all the data 
+under the topics will be saved. Similarly, when a call to save subjects is made, all the data under the subjects
+including topics will be saved. The behavior mentioned is shown in the diagram below.
+
+![subject sq](https://user-images.githubusercontent.com/15065550/97132939-43c3eb80-1783-11eb-91c2-09b79eb2e2f5.png)
+![topic save sq](https://user-images.githubusercontent.com/15065550/97132990-74a42080-1783-11eb-91f8-f6143c0e38c4.png)
+
```
I like how this is sequence diagram is split into 2 so it is neater and easier to read.

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+then the application not only prints the score, but it also prints the incorrect answers from the quiz.
+
+The following diagram shows how you can initiate the quiz for a subject.
+![first](https://user-images.githubusercontent.com/46095141/97313097-3866e200-18a1-11eb-9525-73e38ceb7cbe.png)
```
Should the method name be in quotation marks?

![image](https://user-images.githubusercontent.com/57391861/97386873-c5e11b00-190f-11eb-894b-e2b7fdb69ddb.png)

flsqqq replied:
Should the diagrams be standardised (perhaps by using the same drawing software) to improve consistency?
![image](https://user-images.githubusercontent.com/22489162/97525989-24c39480-19e3-11eb-9c1c-15991bdd2fdd.png)
![image](https://user-images.githubusercontent.com/22489162/97526009-30af5680-19e3-11eb-8a6d-faef11f8d3f6.png)

chuhann replied:
![image](https://user-images.githubusercontent.com/13757965/97530615-78d37680-19ed-11eb-9d2f-520eee7e19c0.png)

1. The method name does not need quotation " ", around it

2. At the end of the lifecycle of "SubjectQuiz", consider removing ":Revised", ":SubjectParser", ":QuizSubjectCommand", ":SubjectQuiz", and ":UI" classes as the classes has already been called earlier. Consider replacing these with "X" to show the end of life line for each classes.

kaijiel24 replied:
![image](https://user-images.githubusercontent.com/60400242/97534055-4bd69200-19f4-11eb-8a2f-57404c498cb9.png)

I think you can omit the use of quotation marks (`""`) for your methods. For other diagrams you managed to do so. (see below)

![image](https://user-images.githubusercontent.com/60400242/97534094-5f81f880-19f4-11eb-82e0-c37131e6479e.png)

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+under the topics will be saved. Similarly, when a call to save subjects is made, all the data under the subjects
+including topics will be saved. The behavior mentioned is shown in the diagram below.
+
+![subject sq](https://user-images.githubusercontent.com/15065550/97132939-43c3eb80-1783-11eb-91c2-09b79eb2e2f5.png)
+![topic save sq](https://user-images.githubusercontent.com/15065550/97132990-74a42080-1783-11eb-91f8-f6143c0e38c4.png)
```
I think it would be good to standardize the way sequence diagrams in the document are drawn.

![image](https://user-images.githubusercontent.com/57391861/97387514-1ad16100-1911-11eb-92f9-e6acda6c489f.png)

![image](https://user-images.githubusercontent.com/57391861/97387534-2755b980-1911-11eb-9334-cd49fb513e93.png)

chuhann replied:
![image](https://user-images.githubusercontent.com/13757965/97528906-be8e4000-19e9-11eb-8447-66c3e2038bcc.png)

Consider changing "sd saving topic" to "alt", "opt", "par" or "loop" as "sd saving topic" is not a valid action

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+* Topic : This segment is accessed through the `topic NAMEOFTOPIC` command. The user can add, delete or find flashcards
+in this segment.
+
+The code is broken down into different packages, which combine classes that perform a similar functionality. Here
+are the list of packages:
```
I think it would be good to add a class diagram to show the structure of the classes.

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+
+The loading of data follows a similar cascading manner as saving, as can be seen in the diagram below.
 
+![subject load sq](https://user-images.githubusercontent.com/15065550/97138326-e59f0480-1792-11eb-8161-7653625d71c4.png)
```
Should there be a class label for this instance ?

![image](https://user-images.githubusercontent.com/57391861/97388649-73096280-1913-11eb-90eb-b0c1be42ba8b.png)

chuhann replied:
![image](https://user-images.githubusercontent.com/13757965/97529935-feeebd80-19eb-11eb-8688-57e70ff4c307.png)

1. Consider changing "ref" to "alt", "opt", "par" or "loop" as "ref" is not a valid action.

2. Self-invocation to class should be a solid arrow instead of a dotted arrow for loadTopics(topicFolders), loadTasks(currentSubjectFolder) and loadResults(resultFile)

xingrong123 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+
+##### Loading data
+
+The loading of data follows a similar cascading manner as saving, as can be seen in the diagram below.
```
Should this be a solid arrow line instead?

![image](https://user-images.githubusercontent.com/57391861/97388929-22463980-1914-11eb-997b-db197c5b092d.png)

xingrong123 reviewed (commented):
Good job! A few mistakes in the sequence diagrams but it was very neat and easy to read.

hermanthong committed:
first ocmit

jialerk committed:
Added sequence diagrams in developer guide

jialerk committed:
added some class diagrams

LIU-YiFeng-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+
+The code is broken down into different packages, which combine classes that perform a similar functionality. Here
+are the list of packages:
+* Card: In charge of classes related to the subject, topic, flashcard and quiz functionalities
+    - Quiz
+        * Quiz
+        * Result
+        * ResultList
+        * TopicQuiz
+        * SubjectQuiz
+    - Flashcard
+    - Subject
+    - SubjectList
+    - Topic
+    - TopicList
+    
```
Would it be better to have an overall design of the application for a clearer illustration?

LIU-YiFeng-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+### Quiz package
+The `Storage` package holds the necessary classes for the quiz functionality of this application. The following are
+the classes in the package:an abstract`Quiz` class, a `Result` class, a `SubjectQuiz` class and a `TopicQuiz` class.
+
+#### Quiz class
+`Quiz` class is an abstract class in the `Quiz` package. It is holds the result of a quiz class and the list of flashcards
+from any subject or topic classes which the user calls the quiz for. Furthermore, it contains a checkAnswer() method the
+checks the answer that the user had given with the correct answer of the quiz. If the user enters the correct answer, the 
+existing score is incremented by one. Else, the contents of the flashcards and the incorrect answer provided by
+the user is transferred to the  `incorrectAnswers` list. Once the user finished the quiz, the application would print the 
+questions that the users did not answer correctly, along the the answer that was provided by the user. 
```
would it be better to have a class diagram? I felt the explanation is a bit wordy.

LIU-YiFeng-1 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+
+#### StorageBuilder
+
+`StorageBuilder` class is a static inner class of the `Storage` class. It follows the builder pattern and its sole purpose
+is to create `Storage` instances. The [builder pattern](https://en.wikipedia.org/wiki/Builder_pattern) is used to deal 
+with the decrease in clarity due to the increase of the number parameters. Every instance of the `Storage` class should 
```
Good use of external resources to guide interested readers to find out more.

flsqqq commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+useful since it lets the students add their tasks. The application then reminds the students
+regarding the upcoming tasks, which would prompt the students to prepare them.
+
+Students who find it difficult to organise woudl find this app useful since it
```
Should it be "would" instead?

LIU-YiFeng-1 reviewed (commented):
Well Done! Easy to read document, I think it would be better if more class diagrams can be included.

flsqqq commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
+4. Import the project as a Gradle project, as described [here](https://se-education.org/guides/tutorials/intellijImportGradleProject.html).
+5. After the importing is complete, locate the `src/main/java/seedu.revised/Revised.java` file, right-click it, and choose
+`Run Revised.main()`.
+6. If the setup is correct, you should see soemthing like this:
```
Should it be "something"? There are occasional typos and grammar mistakes, perhaps they could be fixed to improve readability?

kaijiel24 commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,346 @@
 ## Glossary
 
-* *glossary item* - Definition
+* *Revised* - The main class which initiates the whole application.
```
Perhaps you can link the following items to their respective use case or vice versa (from their use case to glossary). This will help improve the navigability within you DG

CookieHoodie committed:
Replace diagrams with plantuml diagrmas

sugandha929 committed:
edit help date/time format

sugandha929 committed:
edit checkstyle, help

sugandha929 committed:
Merge pull request #87 from CookieHoodie/guide

Replace diagrams with plantuml diagrams

jialerk committed:
Merge branch 'master' into master

jialerk committed:
Merge pull request #88 from sugandha929/master

edit help command

jialerk committed:
Merge branch 'master' into master

jialerk committed:
Edited developer guide

jialerk committed:
Merge branch 'master' of https://github.com/jialerk/tp

# Conflicts:
#	docs/DeveloperGuide.md

jialerk committed:
Edit DG

syncode98 committed:
Merge pull request #86 from jialerk/master

Add diagrams

max-wunan commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,478 @@
+This section describes how RevisED has been designed and implemented so that developers can better understand the 
+designing of the code, with the help of visual aids.
 
-{Describe the value proposition: what problem does it solve?}
+The application has been broken down into different segments so that users would find it easy to navigate from one segment 
+to the next one. There are three segments within the application:
+* Main: This segment edits the list of all the subjects present within the application and takes care of relevant commands 
```
It might be better for you guys to have a diagram for you overall architecture so that the readers can know which classes or interfaces link with/implement each other, instead of trying to understand the overall architecture through the individual sequence/class diagrams below

max-wunan commented on docs/DeveloperGuide.md:
```diff
@@ -1,34 +1,478 @@
+quiz, use the `stop` command. The application will then print out the score that you obtain. If you complete the quiz
+then the application not only prints the score, but it also prints the incorrect answers from the quiz.
+
+The following diagram shows how you can initiate the quiz for a subject.
```
It is a bit cramped here to have a bunch of diagrams with just a single line of explanation while there is no numbering or captions of all the diagrams. As a result it might be a bit confusing for readers. You guys can consider put captions/explanations, at least label with diagram is for what.

CookieHoodie committed:
Corrected syntax error of sequence diagrams

CookieHoodie committed:
Add warning & note symbols and refined storage section

CookieHoodie committed:
Add more info to results.json filename to avoid confusion

CookieHoodie committed:
Fixed deleting of subjects and topics no saved bug

CookieHoodie committed:
Changed behavior of program to proceed with empty data if fails to read from storage

CookieHoodie committed:
Changed sequence diagrams to include clearData & changed result filename

CookieHoodie committed:
Updated documentation to include more storage details

CookieHoodie committed:
Merge branch 'guide' into storage-imp

hermanthong committed:
edit DG

hermanthong committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

# Conflicts:
#	.gitignore
#	docs/DeveloperGuide.md

syncode98 committed:
Merge pull request #122 from CookieHoodie/storage-imp

Improve on storage functionalities and bug handling, storage bug fix

hermanthong committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

hermanthong committed:
fix bugs found in PE

hermanthong committed:
fix whitespace

hermanthong committed:
Fix typo

hermanthong committed:
update gitignore

CookieHoodie committed:
Polished storage part of DG and UG

CookieHoodie committed:
Added more logging docs

hermanthong committed:
trim fullcommand before parsing

hermanthong committed:
Merge branch 'master' of https://github.com/rashien3/tp

CookieHoodie committed:
Added symbol and meaning section to the introduction

CookieHoodie committed:
Removed unnecessary initialization section from UG

CookieHoodie committed:
Refactor the structure of UG

CookieHoodie committed:
Modified user stories to better fit our product

CookieHoodie committed:
Relocated and refined setting up project section

CookieHoodie committed:
Restructured DG

CookieHoodie committed:
Refined testing section

jialerk committed:
Add Junit tests

jialerk committed:
checkstyle compliance

jialerk committed:
Merge pull request #125 from CookieHoodie/guide

Polished storage part of DG and UG

jialerk committed:
Merge pull request #126 from CookieHoodie/guide-restructure

Restructure the feature part of UG

jialerk committed:
Edit test errors

jialerk committed:
Merge pull request #127 from CookieHoodie/dg

Refined DG

jialerk committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

jialerk committed:
Merge branch 'master' of https://github.com/jialerk/tp

jialerk committed:
rename classes

jialerk committed:
Merge branch 'master' of https://github.com/jialerk/tp

hermanthong committed:
Merge branch 'master' into master

syncode98 committed:
Merge pull request #124 from rashien3/master

resolves #116, #117, #119

jialerk committed:
Edit developer guide, added subject and topic command documentation. Changed isExit() of SubjectCommand and TopicCommand to abstract.

jialerk committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

# Conflicts:
#	src/main/java/seedu/revised/card/quizcard/SubjectQuiz.java
#	src/main/java/seedu/revised/command/subjectcommand/AccessSubjectCommand.java
#	src/main/java/seedu/revised/command/subjectcommand/QuizSubjectCommand.java
#	src/main/java/seedu/revised/command/topiccommand/AccessTopicCommand.java
#	src/main/java/seedu/revised/parser/TaskParser.java

jialerk committed:
resolve merge conflict

jialerk committed:
resolve merge conflict 2

CookieHoodie committed:
Merge pull request #128 from jialerk/master

Add Junit tests and added documents in developer guide

CookieHoodie committed:
Restructured DG to follow requirements specified

jialerk committed:
Merge pull request #129 from CookieHoodie/dg

Restructured DG to follow requirements specified

syncode98 committed:
Update the printUpcomingTasks() method

The printUpcomingTasks method will now
only print the task if it is not completed.

CookieHoodie committed:
Made intro more succinct

CookieHoodie committed:
Add more details to logging

syncode98 committed:
Update UserGuide

Add section about how the application
sorts the tasks according to the deadlines.

Update the results section in the other features
section to include an additional note.

Update the results section in the subject and
quiz sections.

CookieHoodie committed:
Modified documentation section

CookieHoodie committed:
Changed format of numbered list

CookieHoodie committed:
Modified and added more non-functional requirements

syncode98 committed:
Update DG for final submission

Restructure the quiz, SubjectQuiz,TopicQuiz
and results classes by splitting them
into design and implementation.

Edit checkstyle errors(Ui class).

syncode98 committed:
Edit UG

Update the headers of certain
sections.

CookieHoodie committed:
Merge pull request #141 from syncode98/DG_updates

Updated both DG and UG

CookieHoodie committed:
Changed example from maths to Cs2113

CookieHoodie committed:
Added more details to manual testing appendix

CookieHoodie committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp into dg

syncode98 committed:
Update UG

Replaced emoji of
the notes icon.

Edited the quiz and results
feature of a subject.

CookieHoodie committed:
Merge pull request #143 from syncode98/DG_updates

Update UG

CookieHoodie committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp into dg

CookieHoodie committed:
Changed NOTE icon in DG

CookieHoodie committed:
Merge pull request #142 from CookieHoodie/dg

Improve on various DG sections

syncode98 committed:
Update DG

Edit DG to ensure that it
it in the first person narrative.

hermanthong committed:
update UG

CookieHoodie committed:
Merge pull request #145 from rashien3/master

update UG, resolves #134

hermanthong committed:
fix bug in list all

CookieHoodie committed:
first draft for portfolio

syncode98 committed:
Merge pull request #146 from rashien3/master

fix bug in list all, solves #144

CookieHoodie committed:
Added list package section under design

syncode98 committed:
Merge pull request #148 from CookieHoodie/dg

Added list package section under design

jialerk committed:
Updated UG documentation

syncode98 committed:
Merge pull request #149 from jialerk/master

Updated UG documentation

syncode98 committed:
Edit DG and UG

syncode98 committed:
Merge branch 'master' into DG_updates

syncode98 committed:
Edit the UG

Edit the command summary to
include more commands.

Included my name beside sections
which I had contributed.

CookieHoodie committed:
Update contribution to DG

hermanthong committed:
add logging, javadoc, fix coding standard violations

hermanthong committed:
change valueOf to parseInt

CookieHoodie committed:
Add more pr to review contribution

CookieHoodie committed:
Merge pull request #150 from syncode98/DG_updates

Dg updates

jialerk committed:
edit User Guide to be more coherent between Subject level and Topic level

jialerk committed:
Merge branch 'master' into master

syncode98 committed:
Update the glossary in DG

syncode98 committed:
Remove exception from application

The RepeatedDateTimeException has been
removed to allow users to add tasks with similar date and time.

hermanthong committed:
Merge pull request #152 from jialerk/master

User Guide to be more coherent between Subject level and Topic level

syncode98 committed:
Merge branch 'master' into DG_updates

jialerk committed:
added javadoc

jialerk committed:
Edit javadoc for subjectcommand classes

syncode98 committed:
Update glossary in DG

CookieHoodie committed:
Merge pull request #151 from rashien3/master

add logging, javadoc, fix coding standard violations

CookieHoodie committed:
Merge pull request #153 from syncode98/DG_updates

Dg updates

syncode98 committed:
Update individual contributions

Edited the page to include contributions.

syncode98 committed:
Edit UG

jialerk committed:
Edit developer guide

sugandha929 committed:
Merge pull request #6 from AY2021S1-CS2113T-W13-1/master

pull changes

syncode98 committed:
Update glossary in DG

syncode98 committed:
Edit the UG.

syncode98 committed:
Edit AboutUs.md file

sugandha929 committed:
Update UG, flashcard print style

hermanthong committed:
Added Untitled Diagram.drawio

CookieHoodie committed:
Merge pull request #154 from syncode98/DG_updates

Dg updates

sugandha929 committed:
Update UserGuide.md

CookieHoodie committed:
Merge branch 'master' into ppp

CookieHoodie committed:
change portfolio link

hermanthong committed:
Update Untitled Diagram.drawio

syncode98 committed:
Merge pull request #155 from sugandha929/master

Update UG (Topic level features) and flashcard print style

hermanthong committed:
Update Untitled Diagram.drawio

hermanthong committed:
Added taskcard.drawio

syncode98 committed:
Merge pull request #156 from CookieHoodie/ppp

Added portfolio

hermanthong committed:
fix typo

hermanthong committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

hermanthong committed:
Delete Untitled Diagram.drawio

hermanthong committed:
Delete taskcard.drawio

hermanthong committed:
Merge pull request #4 from rashien3/remove-drawio

Delete Untitled Diagram.drawio

hermanthong committed:
Merge branch 'master' of https://github.com/rashien3/tp

hermanthong committed:
update DG and list all

CookieHoodie committed:
Update list class diagram

syncode98 committed:
Merge pull request #157 from rashien3/master

Update DG, list all

CookieHoodie committed:
Update portfolio and link to portfolio

jialerk committed:
Added information in DG

CookieHoodie committed:
Merge branch 'master' into dg

jialerk committed:
edit developer guide

jialerk committed:
Merge branch 'master' into master

sugandha929 committed:
Merge pull request #160 from CookieHoodie/dg

Update list class diagram

jialerk committed:
Update FindSubjectCommand.java

jialerk committed:
removed redundant words

jialerk committed:
Merge branch 'master' of https://github.com/jialerk/tp

# Conflicts:
#	docs/DeveloperGuide.md

syncode98 committed:
Modify Ui methods

Certain Ui methods have been
changed to ensure that the application
informs users if they are changing levels.
Previously, the commands were about changing
menues.

jialerk committed:
Merge pull request #158 from CookieHoodie/ppp

Update portfolio and link to portfolio

CookieHoodie committed:
Merge pull request #159 from jialerk/master

Added command documentation in DG and deleted irrelevant classes

CookieHoodie committed:
Update exportcommand description

hermanthong committed:
update design of card package

sugandha929 committed:
Merge pull request #161 from CookieHoodie/dg

Update exportcommand description

sugandha929 committed:
Merge pull request #7 from AY2021S1-CS2113T-W13-1/master

pull changes

syncode98 committed:
Edit the DG

Edit the glossary section of the DG to
include more methods. Sort the
contents alphabetically.

Updated the individual PPP.

syncode98 committed:
Merge branch 'master' into DG_updates

hermanthong committed:
add architecture

hermanthong committed:
Merge branch 'master' into master

sugandha929 committed:
Merge pull request #162 from rashien3/master

add architecture and update class package in DG

syncode98 committed:
Edit documentations

syncode98 committed:
Edit Ui class

Minor adjustments in the printUpcomingtask
method in the Ui class.

hermanthong committed:
Merge pull request #163 from syncode98/DG_updates

Dg updates

jialerk committed:
Beginning to add FlashcardCommand Test and TaskCommand Test

sugandha929 committed:
Merge pull request #8 from AY2021S1-CS2113T-W13-1/master

pull changes

syncode98 committed:
Edit DG

syncode98 committed:
Merge branch 'master' into DG_updates

syncode98 committed:
Minor changes to DG

syncode98 committed:
Edit the DG

Edit the results section in the DG

hermanthong committed:
update DG

CookieHoodie committed:
Merge pull request #165 from rashien3/master

update DG

sugandha929 committed:
Update DG, add javadoc comments, update subjectparser

jialerk committed:
Added and edited test cases for command class

jialerk committed:
Added an exception to resolve issue #166

jialerk committed:
Remove redundant information not used in code

jialerk committed:
Fixed grammatical error

sugandha929 committed:
Merge branch 'master' into master

hermanthong committed:
fix error messages

sugandha929 committed:
edit checkstyle

sugandha929 committed:
Merge branch 'master' of https://github.com/sugandha929/tp

hermanthong committed:
fix error message

jialerk committed:
Fixed error whereby using unicode in tests cannot pass java checkstyle

hermanthong committed:
fix checkstyle

CookieHoodie committed:
Changed expected output to sample output

sugandha929 committed:
Merge pull request #164 from syncode98/DG_updates

Dg updates

sugandha929 committed:
Update DeveloperGuide.md

sugandha929 committed:
Merge pull request #169 from rashien3/master

fix error messages, resolves #95, #96

hermanthong committed:
Merge pull request #170 from CookieHoodie/ug

Changed expected output to sample output

jialerk committed:
Changed project to product

syncode98 committed:
Remove the timing feature

The timing feature fails
for certain cases. Hence, it has been
removed.

sugandha929 committed:
Merge branch 'master' into master

sugandha929 committed:
Merge pull request #168 from jialerk/master

Added Test cases

syncode98 committed:
Merge Master

hermanthong committed:
Merge pull request #167 from sugandha929/master

update dg, javadoc, subjectParser

syncode98 committed:
Merge branch 'master' into DG_updates

jialerk committed:
made the UG more coherent

syncode98 committed:
Merge pull request #171 from jialerk/master

made the UG more coherent

syncode98 committed:
Merge branch 'master' into DG_updates

sugandha929 committed:
Update UserGuide.md

syncode98 committed:
Merge pull request #172 from syncode98/DG_updates

Dg updates

hermanthong committed:
Merge pull request #173 from sugandha929/master

Update UserGuide.md

hermanthong committed:
remove unicode

hermanthong committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

# Conflicts:
#	docs/UserGuide.md

sugandha929 committed:
Merge pull request #9 from AY2021S1-CS2113T-W13-1/master

pull changes

hermanthong committed:
add format for tasks

CookieHoodie committed:
Merge pull request #175 from rashien3/master

remove smiley face, tick and cross unicode characters

sugandha929 committed:
add FlashcardCommand test case

hermanthong committed:
Merge pull request #177 from sugandha929/master

add flashcardcommand test case

syncode98 committed:
Edit userguide and portfolio

syncode98 committed:
Merge branch 'master' into DG_updates

syncode98 committed:
Edit UG to rectified mistakes

syncode98 committed:
Edit individual file

hermanthong committed:
edit code and UG to make sample output consistent

sugandha929 committed:
Update PPP

sugandha929 committed:
Update sugandha.md

sugandha929 committed:
Merge branch 'master' into master

sugandha929 committed:
Merge pull request #180 from syncode98/DG_updates

Dg updates

hermanthong committed:
fix find under subject level

syncode98 committed:
Merge pull request #179 from sugandha929/master

Update PPP

syncode98 committed:
Update individual portfolio

jialerk committed:
Removed some words in DG

jialerk committed:
Added my own PPP

sugandha929 committed:
Merge pull request #178 from rashien3/master

edit code and UG to make sample output consistent

syncode98 committed:
Merge branch 'master' into DG_updates

sugandha929 committed:
update ppp name

sugandha929 committed:
Merge branch 'master' of https://github.com/sugandha929/tp

jialerk committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

hermanthong committed:
fix checkstyle

sugandha929 committed:
Merge pull request #183 from rashien3/master

fix checkstyle

sugandha929 committed:
Update sugandha929.md

jialerk committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp

CookieHoodie committed:
Merge pull request #181 from jialerk/master

Did my own PPP

syncode98 committed:
Update portfolio

CookieHoodie committed:
Merge pull request #182 from sugandha929/master

update ppp name

hermanthong committed:
Merge pull request #184 from syncode98/DG_updates

Updated portfolio

syncode98 committed:
Edit the portfolio

syncode98 committed:
Updated the individual portfolio

syncode98 committed:
Delete TaskList.java

hermanthong committed:
Merge pull request #185 from syncode98/DG_updates

Edit the portfolio

syncode98 committed:
update url

syncode98 committed:
Merge branch 'master' of https://github.com/AY2021S1-CS2113T-W13-1/tp into DG_updates

syncode98 committed:
Merge pull request #186 from syncode98/DG_updates

Edit aboutus.md
